### ISSUE TITLE
Headless Pi auto install issues, crashes PINN

### ISSUE THREAD

**Issue by noafterglow** [procount/pinn#414]
It appears that specifying no_default_source crashes PINN with vncshare enabled.

This does not happen immediately though, it waits for the network to come up first, which seems to take a REALLY long time (30 seconds).

If no_default_source is specified, then the screen blanks, any accumulated messages show on top, and then "Recovery application crashed" appears, along with a command prompt.

Note that I have also been able to crash pinn without vncshare,  it looks like not having a remote os list is the cause of the crash.

**Comment by procount** [comment]
Woah, really? I must look into that one! Never seen that before. Thanks for reporting.

**Comment by procount** [comment]
Well, I tried this. I had no_default_source, vncshare, I removed the repo option, had no USBs fitted, and it worked ok with no crash. I did have an OS already installed. I think you'll need to provide more info, such as: model of Pi, version of PINN, contents of recovery.cmdline, exact procedure to make it happen, type of network connection, maybe even a dump of the debug file (see wiki troubleshooting) and anything else that might be different to my setup. Because I can't make it crash.
